lib/compat.php

```diff
@@ -77,6 +77,84 @@ function gutenberg_provide_render_callback_with_block_object( $pre_render, $bloc
 }
 add_filter( 'pre_render_block', 'gutenberg_provide_render_callback_with_block_object', 10, 2 );
 
+/**
+ * Registers the user meta property responsible for storing preferences for the
+ * data script persistence middleware. The meta is explicitly registered so that
+ * it is made accessible for read and write via the REST API.
+ */
+function gutenberg_register_data_persistence_user_meta() {
+	global $wpdb;
+	register_meta(
+		'user',
+		$wpdb->prefix . 'data_persistence',
+		array(
+			'type'         => 'object',
+			'single'       => true,
+			'show_in_rest' => array(
+				'name'   => 'data_persistence',
+				'type'   => 'object',
+				'schema' => array(
+					'type'                 => 'object',
```

Comment:
Should a default value be useful here?

Reply:
> Should a default value be useful here?

Would you expect it to have an impact? Or what would you hope should happen by providing a default?

My concern with this might be: We don't want to set a meta value unless one is actually provided with the request.

Trying to follow how this logic flows in the REST API implementation, it doesn't appear it should have an impact one way or the other (at least currently), since it seems like the schema defined here is [referenced](https://github.com/WordPress/WordPress/blob/77aa5d1604d74acb67fe6954e66670f234e0313c/wp-includes/rest-api/fields/class-wp-rest-meta-fields.php#L151) only after [considering that a value was provided](https://github.com/WordPress/WordPress/blob/77aa5d1604d74acb67fe6954e66670f234e0313c/wp-includes/rest-api/fields/class-wp-rest-meta-fields.php#L137-L139).